pkg/mimirpb/custom.go

```diff
@@ -8,8 +8,76 @@ import (
 	"math"
 
 	"github.com/prometheus/prometheus/model/histogram"
+	"google.golang.org/grpc/encoding"
+	"google.golang.org/grpc/encoding/proto"
+	"google.golang.org/grpc/mem"
+	protobufproto "google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/protoadapt"
 )
 
+func init() {
+	c := encoding.GetCodecV2(proto.Name)
+	encoding.RegisterCodecV2(&CodecV2{c})
+}
+
+// CodecV2 customizes gRPC unmarshalling.
+type CodecV2 struct {
```

Comment:
I would rename this codec to reflect it's purpose, something like: `BufferHolderCodec`

Reply:
I think the name `CodecV2` makes complete sense, since it wraps the `codecV2` type in `grpc/encoding/proto` and it implements the `grpc/encoding.CodecV2` interface.

Reply:
IMO, our codec, that does special things, should reflect in it's name what's the special thing it's doing. Their `codecV2` name is because it's the default implementation.

BTW, do we need ours to be exported?